server/service/src/invoice/outbound_shipment/update/generate.rs

```diff
@@ -157,7 +157,8 @@ fn lines_to_trim(
     let mut empty_lines = InvoiceLineRepository::new(connection).query_by_filter(
         InvoiceLineFilter::new()
             .invoice_id(EqualFilter::equal_to(&invoice.id))
-            .number_of_packs(EqualFilter::equal_to_f64(0.0)),
+            .number_of_packs(EqualFilter::equal_to_f64(0.0))
+            .r#type(InvoiceLineRowType::StockOut.equal_to()),
```

Comment:
This would fix not including service charge calculation in the grand total but would appear as dummy line in the ci of the msupply cause of pack size, no. of packs with other price fields are being 0 (of the service charge item) in the oms. Does that mean the service charge feature only need to be in oms (not in the msupply)? What is the reason behind mapping service charge (in oms) to the service item (in the msupply) though?

In the msupply, service item has default pack size to 1 and it uses default sell price in the ci. Above all what is the expected behavior for service charge both in the oms (created from) and the ms (received after sync)?

Sorry I may not have the proper context to do any further changes apart from the current minimal change.

Reply:
I agree with all your questions and conclusions, so I am happy with this fix for now!